erpnext/public/js/education/lms/quiz.js

```diff
@@ -43,6 +55,37 @@ class Quiz {
 		else {
 			this.make_actions();
 		}
+		window.addEventListener('beforeunload', (event) => {
+			event.preventDefault();
+			event.returnValue = '';
+		});
+	}
+
+	set_timer(duration) {
+		this.time_left = duration;
+		var self = this;
+		this.calculate_and_display_time(this.time_left, "Time Left - ");
+		this.timer = setInterval(function () {
+			self.time_left -= 1;
```

Comment:
the set interval timer is not reliable imo
It schedules the function but executes when it gets chance
Instead of incrementing decrementing the time left, using the inbuilt date,time function could be better

```
var TimeCapture = function () {
    var start = new Date().getTime();
    var last = start;
    var now = start;
    this.start = function () {
        start = new Date().getTime();
    };
    this.check = function (message) {
        now = (new Date().getTime());
        console.log(message, 'START:', now - start, 'LAST:', now - last);
        last = now;
    };
};

//Example:
var time = new TimeCapture();
//begin tracking time
time.start();
//...do stuff
time.check('say something here')//look at your console for output
//..do more stuff
time.check('say something else')//look at your console for output
//..do more stuff
time.check('say something else one more time')//look at your console for output
```